### DISCUSSION TITLE
Event On isDragging Ended

### DISCUSSION THREAD

**Discussion by AndrewPixel** [ThibaultJanBeyer/DragSelect#138] [Q&A] [answered]
I asked "Is there any event which lets me know when the user has stopped dragging the selected elements eg something like isDraggingEnd ?"

And you responded "Yes for sure, if you check the docs for [callbacks](https://github.com/ThibaultJanBeyer/DragSelect#event-callbacks), there is callback which is triggered on every end of interaction. So in your case you’d look for true on the isDragging callback object key." but I don't follow :-(

I've created a dragend event and added it to your _defineProperty for _reset

```
      _defineProperty(this, "_reset", function (event) {
        var isDragging = _this.isDragging;

        if (isDragging) {  // new code
          _this.DS.publish('Interaction:dragend', { });
        }

        _this.stop();
```
to get what I need. That is, when on mouse up, following a dragging selection attempt, a single event is thrown.

**Comment by ThibaultJanBeyer** [comment, marked as answer]
Hi Andrew, lovely to hear from you!

No need for a custom event.
As mentioned, you can use the event called `callback` to know when the user stopped dragging the selected elements.

Here is the pseudocode:

```JavaScript
const ds = new DragSelect({
  selectables: document.querySelectorAll('.item')
});

ds.subscribe('callback', ({ isDragging, items }) => {
  if(isDragging) {
    console.log("user has stopped dragging the selected elements:", items)
  }
})
```
It will only console.log after the user stopped dragging. You can [see this in practice in this codepen](https://codepen.io/ThibaultJanBeyer/pen/mdLGdZr?editors=0011).

Cheers 🍻

**Comment by AndrewPixel** [comment]
I didn't think that would work till I tried it. Thank you (I've now removed the code I added.)
